private/bufpkg/bufmodule/bufmodulestore/module_data_store.go

```diff
@@ -129,9 +138,7 @@ func (p *moduleDataStore) GetModuleDatasForModuleKeys(
 	for _, moduleKey := range moduleKeys {
 		moduleData, err := p.getModuleDataForModuleKey(ctx, moduleKey)
 		if err != nil {
-			if !errors.Is(err, fs.ErrNotExist) {
-				return nil, nil, err
-			}
+			// Treat any error returned as a cache miss
```

Comment:
This seems wrong - I'd want to understand why we are changing this

Reply:
In the previous implementation, we are burying the errors even deeper:
- We coerce a call on `deleteInvalidModuleData` on every error https://github.com/bufbuild/buf/blob/8996858e1193b1918fdc328d73805eb35a6acd8b/private/bufpkg/bufmodule/bufmodulestore/module_data_store.go#L176-L180
- And we only surface an error from the `deleteInvalidModuleData` process, otherwise we then coerce all of those errors to be a `PathError` https://github.com/bufbuild/buf/blob/8996858e1193b1918fdc328d73805eb35a6acd8b/private/bufpkg/bufmodule/bufmodulestore/module_data_store.go#L277-L286
- And then all of those path errors are then considered a cache miss

So there are a couple of things about this change:
- We are simply returning the error here and transparently treating it as a cache miss -- if we want, we can log these errors
- We removed `deleteInvalidModuleData`, which causes cache corruption because it was creating a race